Security/Firewall/GetJWTListener.php

```diff
@@ -98,26 +97,34 @@ public function handle(GetResponseEvent $event)
     {
         $request = $event->getRequest();
 
-        if ($this->options['post_only'] && !$request->isMethod('POST')) {
-            $event->setResponse(new JsonResponse('invalid method', 405));
+        if (!in_array($request->getMethod(), ['POST', 'GET'])) {
             return;
         }
 
-        if ($this->options['post_only']) {
-            $username = trim($request->request->get($this->options['username_parameter'], null, true));
-            $password = $request->request->get($this->options['password_parameter'], null, true);
+        if ($request->isMethod('POST')) {
+            if ('json' === $request->getContentType()) {
+                $params = json_decode($request->getContent(), true);
+                if (JSON_ERROR_NONE !== json_last_error()) {
+                    throw new BadRequestHttpException('Bad JSON request.');
```

Comment:
I believe this should be a `Symfony\Component\Security\Core\Exception\AuthenticationException` too.